### ISSUE TITLE
Differential tests do not recognise groups with zero reads

### ISSUE THREAD

**Issue by Trenthaydon** [bryandmartin/corncob#100]
Hi,

I noticed when running differential tests with corncob that it does not recognise any significant differences between groups when there is an ASV which is highly abundant in one group but not present at all in the other group. Is there any way around this? 

thanks,
Trent

**Comment by bryandmartin** [comment]
Hi Trent,

Could you give an example of the kind of hypothesis you are interested in testing? In this case, you have no data in one group, so you can't estimate parameters associated with that group. corncob explicitly filters for that, you can see the details in the documentation, check out `filter_discriminant` for example.

If the group is of primary interest, you can try switching from Wald to LRT in the `test` option. However, this will come with the disclaimer that you have to be careful on fitting models for empty groups!

**Comment by Trenthaydon** [comment]
Hi Bryan,

thanks for getting back to me.

I'm looking at the differences in bacterial communities in a coral species between two contrasting environments. There are significant differences between the overall communities of this one species of coral between both environments, so I'm trying to use corncob to look at any differentially abundant taxa between the two sites. I'm running this on rarified data at both the ASV level and also genus level. Besides the issues with the dispertion of the data across replicates, I have noticed that there are a lot of ASVs/genera only found in one environment and not the other and are therefore not being picked up by this analysis.

Perhaps this is not the most appropriate test for my data. I also tried restricting the filter_discriminant and the LRT test instead, but each time I get zero significant taxa when running this test. When looking at the actual p-values of individual ASVs (da_analysis$p) there are some that are significant, but i'm presuming they are discarded when the FDR is applied. I might try reducing the number of ASVs in the analysis (i.e. filtering out much of the low abundant genera/ASVs) and then running it. As for taxa not present in one group, I may have to add the addition of another type of analysis for this.

One other thought, this particular example I gave is only a subsample of the dataset. The phyloseq object likely contains thousands of ASVs not present in any of the two groups but remain in the sample subset phyloseq object. Is it possible that this may also effect the outcome of the analysis?

thanks,
Trent

________________________________
